src/main/java/teammates/ui/webapi/action/LinkRecoveryAction.java

```diff
@@ -0,0 +1,87 @@
+package teammates.ui.webapi.action;
+
+import teammates.common.exception.EmailSendingException;
+import teammates.common.exception.TeammatesException;
+import teammates.common.util.Const;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.Logger;
+import teammates.logic.api.EmailGenerator;
+import teammates.ui.webapi.output.ApiOutput;
+
+
+/**
+ * Action specifically created for confirming email and sending recovery link.
+ */
+public class LinkRecoveryAction extends Action {
+
+    private static final Logger log = Logger.getLogger();
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.PUBLIC;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        // no specific access control needed.
+    }
+
+    @Override
+    public ActionResult execute() {
+        String requestedEmail = getNonNullRequestParamValue(Const.ParamsNames.RESTORE_EMAIL);
+        boolean hasStudentsWithRestoreEmail = !logic.getAllStudentForEmail(requestedEmail).isEmpty();
+
+        if (hasStudentsWithRestoreEmail) {
+
+            EmailWrapper email = new EmailGenerator().generateLinkRecoveryEmail(requestedEmail);
+
+            if (email != null) {
+                try {
+                    emailSender.sendEmail(email);
+                } catch (EmailSendingException e) {
+                    log.severe("Link recovery email failed to send.: "
+                            + TeammatesException.toStringWithStackTrace(e));
+                    return new JsonResult(new EmailRestoreResponse(EmailResponseResult.SUCCESS_BUT_EMAIL_FAIL_TO_SEND,
+                            "Link recovery email failed to send."));
+                }
+            }
+
+            return new JsonResult(new EmailRestoreResponse(EmailResponseResult.SUCCESS,
+                    "An recovery link has just been sent to the given email."));
```

Comment:
`The recovery links for your feedback sessions have been sent to the specified email.` instead?